### PR TITLE
CORE-538, CORE-544 Handle move block action

### PR DESCRIPTION
**Pull request by object-required**
[Screencast from 2023-05-29 04:24:40 PM.webm](https://github.com/FLOIP/flow-builder/assets/3971957/0123ea66-3d7d-4874-89d8-ab88c0b71486)

### CODE REVIEW

File: src/components/common/PlainDraggable.vue [hunk 1]
```diff
@@ -35,11 +35,29 @@ export class PlainDraggable extends mixins(Lang) {
...
+    // this.draggable.snap = {x: 50, y: 50, width: 50, height: 50}
+    // this.draggable.snap = {step: 21}
+
+    this.handleInitialized()
+    // })
+  }
+
+  updated(): void {
+    this.draggable.setOptions(this.plainDraggableLibOptions)
+  }
```
Comment (object-required) [inline src/components/common/PlainDraggable.vue:53]: `<Block>` passes updated coordinates to `<PlainDraggable startX startY>` which allows us to update block position when needed.

File: src/store/builder/index.ts [hunk 2]
```diff
@@ -309,6 +304,14 @@ export const actions: ActionTree<IBuilderState, IRootState> = {
...
     return newFlowContainer ?? undefined
   },
+
+  setBlockPositionTo({commit, rootGetters}, {position: {x, y}, block}) {
+    commit('flow/flow_updateBlockCoordinates', {
+      flowId: rootGetters['flow/activeFlow'].uuid,
+      blockId: block.uuid,
+      coordinates: {x, y},
+    }, {root: true})
+  },
```
Comment (object-required) [inline src/store/builder/index.ts:314]: Making this an action allows committing a mutation in the neighbor `flow` module.

File: src/store/builder/index.ts [hunk 3]
```diff
@@ -140,11 +140,6 @@ export const mutations: MutationTree<IBuilderState> = {
     operations[operation.kind] = operation
   },
 
-  setBlockPositionTo(state, {position: {x, y}, block}) {
-    block.ui_metadata.canvas_coordinates.x = x
-    block.ui_metadata.canvas_coordinates.y = y
-  },
```
Comment (object-required) [inline src/store/builder/index.ts:146]: Moved to `flow/flow_updateBlockCoordinates` to trigger a mutation that `undoRedo` can handle.

File: src/store/flow/flow.ts [hunk 4]
```diff
@@ -202,6 +202,21 @@ export const mutations: MutationTree<IFlowsState> = {
...
+    const flow: IFlow = findFlowWith(flowId, state as unknown as IContext)
+    const block: IBlock = findBlockWith(blockId, flow)
+
+    if (!block.ui_metadata) {
+      Vue.set(block, 'ui_metadata', {
+        canvas_coordinates: coordinates,
+      })
+    } else {
+      Vue.set(block.ui_metadata, 'canvas_coordinates', coordinates)
+    }
```
Comment (bulatgab) [inline src/store/flow/flow.ts:218]: A theoretical question @object-required @safydy-r 
What would happen if we do:
```
Vue.set(block, 'ui_metadata', {canvas_coordinates: coordinates})
```
without if-else? I mean what would happen in terms of Vue reactivity if `ui_metadata.canvas_coordinates` exists? What's the danger of vue-setting a higher level object?

I think it's possible it React, for example, right? Even preferrable, I think.
Do you know how Vue handles this? I couldn't find any article on this (maybe I was googling it wrong) - could you help, please?

**Review by bulatgab — APPROVED** [review]
Clean refactoring/fix, thank you.

Left an off-topic question.

File: src/store/builder/index.ts (hunk 3 above)
Comment (safydy-r) [reply src/store/builder/index.ts:146]: Nice solution, I agree

**Review by safydy-r — APPROVED** [review]
Joining @bulatgab 's comment, this is a clean one. Thanks @object-required 👍

**Pull request merged**